testsuite/Properties.hs

```diff
@@ -0,0 +1,30 @@
+module Main (
+    main
+  ) where
+
+import Test.Tasty
+import Test.Tasty.QuickCheck as QC
+
+import Morte.Core
+import ClosedWellTyped
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Properties" [
+      QC.testProperty "normalize . normalize == normalize" normalizeIdempotent 
+    , QC.testProperty "Well typed after normalization"     wellTypedAfterNormalize
```

Comment:
Add one more property to test if inferred types are already in normal form like you asked about previously.  In other words:

```haskell
fmap normalize . typeOf = typeOf
```